Rebuilding from a project source directory fails with .NET Core's msbuild

git clean and building with:
```
build.cmd -subsetCategory coreclr -c Release && build.cmd -subsetCategory libraries /p:CoreCLRConfiguration=Release
```

And then building from within VS works great. But when building projects via command line, I get errors.

From within a source directory:

```
C:\dev\runtime-lowlevel\src\libraries\System.Net.Http\src>dotnet msbuild /T:Rebuild
Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Users\conelso\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\conelso\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\dev\runtime-lowlevel\src\libraries\System.Net.Http\src\System.Net.Http.csproj]
```

From within a test directory I get similar error, and it also wipes out Corefx.Private.TestUtilities which breaks VS going forward until I call build.cmd again.
```
C:\dev\runtime-lowlevel\src\libraries\System.Net.Http\tests\FunctionalTests>dotnet msbuild /T:RebuildAndTest
Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Users\conelso\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\conelso\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\dev\runtime-lowlevel\src\libraries\Common\tests\CoreFx.Private.TestUtilities\CoreFx.Private.TestUtilities.csproj]
```

When using the VS version of msbuild, aka Developer Command Prompt and `msbuild` instead of `dotnet msbuild`, this failure doesn't happen.

CC @joperezr @safern

--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "3.0.100",
+    "version": "3.1.101",
     "allowPrerelease": true,
     "rollForward": "major"
   },

